src/webots/vrml/WbNode.cpp

```diff
@@ -972,13 +974,43 @@ void WbNode::writeParameters(WbVrmlWriter &writer) const {
     parameter->write(writer);
 }
 
+bool WbNode::isUrdfLinkRoot() const {
+  return findSFString("name") ? true : false;
+}
+
+WbNode *WbNode::findUrdfLinkRoot() const {
+  WbNode *parentRoot = parent();
+  while (!parentRoot->isUrdfLinkRoot())
+    parentRoot = parentRoot->parent();
+  return parentRoot;
+}
+
 void WbNode::write(WbVrmlWriter &writer) const {
   if (uniqueId() == -1) {
     if (nodeModelName() == "Plane" || nodeModelName() == "Capsule") {
       WbNodeFactory::instance()->exportAsVrml(this, writer);
       return;
     }
   }
+  if (writer.isUrdf()) {
+    if (gUrdfCurrentNode != this && nodeModelName().contains("Joint") && !gUrdfNodesQueue.contains(this)) {
```

Comment:
> nodeModelName().contains("Joint")

That looks dangerous (if for example in the future we add a new join who doesn't contain the 'Joint' keyword, we might miss it here without noticing it). But I don't see any better solution :-/

Reply:
We can do the same thing as for Robot (`isRobot()`), we can add `isJoint()` method and override it in `WbBasicJoint`.